x-pack/plugins/lens/public/react_embeddable/expressions/on_render.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaExecutionContext } from '@kbn/core-execution-context-common';
+import { canTrackContentfulRender } from '@kbn/presentation-containers';
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
+import { TableInspectorAdapter } from '../../editor_frame_service/types';
+
+import { getExecutionContextEvents, trackUiCounterEvents } from '../../lens_ui_telemetry';
+import { GetStateType, LensApi, LensEmbeddableStartServices, LensInternalApi } from '../types';
+import { getSuccessfulRequestTimings } from '../../report_performance_metric_util';
+import { addLog } from '../logger';
+
+function trackContentfulRender(activeData: TableInspectorAdapter, parentApi: unknown) {
+  if (!activeData || !canTrackContentfulRender(parentApi)) {
```

Comment:
`activeData` exists by definition, if that wasn't intended better to change the function argument type